### PR TITLE
Update API tags to make beta methods public (background map, Bing services, solar calculations).

### PR DESCRIPTION
**Pull request by RBBentley**

### CODE REVIEW

File: common/api/imodeljs-frontend.api.md [hunk 1]
```diff
@@ -1562,33 +1562,25 @@ export interface BeWheelEventProps extends BeButtonEventProps {
...
-    // (undocumented)
-    getHeights(range: Range2d): Promise<any>;
-    // (undocumented)
     getHeightValue(point: Point3d, iModel: IModelConnection, geodetic?: boolean): Promise<number>;
     // (undocumented)
     protected _requestContext: ClientRequestContext;
     }
 
-// @internal (undocumented)
+// @internal
```
Comment (pmconne) [inline common/api/imodeljs-frontend.api.md:1580]: It looks like you meant to change this to `public`, right? This API is very useful - for example, I am using it to [demonstrate](https://github.com/imodeljs/frontend-sample-showcase/pull/161) how to navigate to a named location on the globe.

**Review by pmconne — CHANGES_REQUESTED** [review]
PR description suggests solar calculation APIs have been promoted but it appears they have not been.

File: core/frontend/src/tile/map/BingElevation.ts [hunk 2]
```diff
@@ -46,6 +48,10 @@ export class BingElevationProvider {
     this._seaLevelOffsetRequestTemplate = "https://dev.virtualearth.net/REST/v1/Elevation/SeaLevel?points={points}&key={BingMapsAPIKey}".replace("{BingMapsAPIKey}", bingKey);
     this._heightListRequestTemplate = "https://dev.virtualearth.net/REST/v1/Elevation/List?points={points}&heights={heights}&key={BingMapsAPIKey}".replace("{BingMapsAPIKey}", bingKey);
   }
+
+  /** Return the height (altitude) at a given cartographic location
```
Comment (pmconne) [inline core/frontend/src/tile/map/BingElevation.ts:52]: Can you explain what `geodetic` does here and elsewhere?

File: core/frontend/src/BingLocation.ts [hunk 3]
```diff
@@ -27,6 +29,9 @@ export class BingLocationProvider {
     this._locationRequestTemplate = "https://dev.virtualearth.net/REST/v1/Locations?query={query}&key={BingMapsAPIKey}".replace("{BingMapsAPIKey}", bingKey);
     // this._localCircularSearchRequestTemplate = "https://dev.virtualearth.net/REST/v1/LocalSearch/?CircularMapView={{circle}}&key={BingMapsAPIKey}".replace("{BingMapsAPIKey}", bingKey);
   }
+  /** Return the location of a query (or undefined if not found)
```
Comment (pmconne) [inline core/frontend/src/BingLocation.ts:32]: Can you elaborate on the expected contents of "query"?

**Review by pmconne — CHANGES_REQUESTED** [review]

**Comment by RBBentley** [comment]
> PR description suggests solar calculation APIs have been promoted but it appears they have not been.

Fixed.

File: core/frontend/src/tile/map/BingElevation.ts (hunk 2 above)
Comment (RBBentley) [reply core/frontend/src/tile/map/BingElevation.ts]: Yes, done.

File: common/api/imodeljs-frontend.api.md (hunk 1 above)
Comment (RBBentley) [reply common/api/imodeljs-frontend.api.md:1580]: Yes.  Fixed.

File: core/frontend/src/BingLocation.ts (hunk 3 above)
Comment (RBBentley) [reply core/frontend/src/BingLocation.ts]: Yes, done.

File: common/api/imodeljs-frontend.api.md (hunk 1 above)
Comment (RBBentley) [reply common/api/imodeljs-frontend.api.md:1580]: Yes, fixed.

Comment (pmconne) [reply common/api/imodeljs-frontend.api.md:1580]: It still says internal - did you rerun extract-api?

File: core/frontend/src/BackgroundMapGeometry.ts [hunk 4]
```diff
@@ -364,7 +364,10 @@ export class BackgroundMapGeometry {
   }
 }
 
-/** @internal */
+/** Calculate the ECEF to database (IModel) coordiante transform at a provided location, using the GCS of the iModel.
```
Comment (pmconne) [inline core/frontend/src/BackgroundMapGeometry.ts:367]: /s/coordiante/coordinate

File: core/frontend/src/tile/map/BingElevation.ts [hunk 5]
```diff
@@ -46,6 +48,11 @@ export class BingElevationProvider {
     this._seaLevelOffsetRequestTemplate = "https://dev.virtualearth.net/REST/v1/Elevation/SeaLevel?points={points}&key={BingMapsAPIKey}".replace("{BingMapsAPIKey}", bingKey);
     this._heightListRequestTemplate = "https://dev.virtualearth.net/REST/v1/Elevation/List?points={points}&heights={heights}&key={BingMapsAPIKey}".replace("{BingMapsAPIKey}", bingKey);
   }
+
+  /** Return the height (altitude) at a given cartographic location.
+   * If [[geodetic]] is true (the default) then height is returned in the Ellipsoidal WGS84 datum.  If [[geodetic]] is false then the sea level height id returned using the Earth Gravitational Model 2008 (EGM2008 2.5’).
```
Comment (pmconne) [inline core/frontend/src/tile/map/BingElevation.ts:53]: [[geodetic]] is not a valid link.

File: core/frontend/src/BackgroundMapGeometry.ts (hunk 4 above)
Comment (RBBentley) [reply core/frontend/src/BackgroundMapGeometry.ts]: Fixed

File: core/frontend/src/tile/map/BingElevation.ts (hunk 5 above)
Comment (RBBentley) [reply core/frontend/src/tile/map/BingElevation.ts]: Fixed

File: common/api/imodeljs-frontend.api.md (hunk 1 above)
Comment (RBBentley) [reply common/api/imodeljs-frontend.api.md]: Neglected to extract-api.  Fixed.

**Review by pmconne — APPROVED** [review]

File: common/api/presentation-components.api.md [hunk 6]
```diff
@@ -167,10 +167,10 @@ export interface DataProvidersFactoryProps {
 export const DEFAULT_PROPERTY_GRID_RULESET: Ruleset;
 
 // @beta @deprecated
-export function DEPRECATED_controlledTreeWithFilteringSupport<P extends ControlledTreeWithVisibleNodesProps>(TreeComponent: React.FC<P>): React.FC<Pick<P & ControlledTreeWithFilteringSupportProps, "filter" | "onFilterApplied" | "onMatchesCounted" | "activeMatchIndex" | "nodeLoader" | "onNodeLoaderChanged" | Exclude<keyof P, "visibleNodes">>>;
```
Comment (pmconne) [inline common/api/presentation-components.api.md:170]: Revert this file. `rush extract-api` is terrible. I usually just run `npm run extract-api` in the directories of the packages I modified.

Comment (RBBentley) [reply common/api/presentation-components.api.md]: Fixed. Thanks.   I usually discard the changes before I commit but running extract-api in directories sounds more reliable.

Comment (RBBentley) [reply common/api/presentation-components.api.md]: And then somehow the merge re-added a bunch of changes.   I'll have to complete this next week.

**Pull request merged**